crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -417,6 +420,30 @@ impl Timeline {
         Ok(())
     }
 
+    pub async fn edit_poll(
+        &self,
+        fallback_text: impl Into<String>,
+        poll: UnstablePollStartContentBlock,
+        edit_item: &EventTimelineItem,
+    ) -> Result<(), UnsupportedEditItem> {
+        // Early returns here must be in sync with
+        // `EventTimelineItem::can_be_edited`
+        let Some(event_id) = edit_item.event_id() else {
+            return Err(UnsupportedEditItem::MISSING_EVENT_ID);
+        };
+        let TimelineItemContent::Poll(_) = edit_item.content() else {
+            return Err(UnsupportedEditItem::NOT_POLL_EVENT);
+        };
```

Comment:
On the one hand, `can_be_edited` requires that the `edit_item.content()` be of type `Message`, not `Poll`, so the two won't work nicely together.

On the other hand, `can_be_edited` isn't even exposed to the FFI layer; it's only used in tests. So, we can do one of two things:

- remove the comment; trying to misuse this function will result in an `Err` result, which carries the right semantics here.
- introduce a `can_poll_be_edited` method with the right preconditions, and expose it to the FFI layer.

I'm fine going with the former here, since the extra check function would be superfluous with respect to the checks done here anyways.